### PR TITLE
Adapt install & upgrade script for Debian Buster (#26)

### LINKED ISSUE #26: [Buster] Adapt the PHP 7.x custom install / requirement
c.f. https://ci-buster.nohost.me/ci/job/416

Following the discussion started [here](https://github.com/YunoHost/issues/issues/1255#issuecomment-596254890), this app needs to be adapted both to : 
- be installable on a fresh Buster system
- make the transition smooth for people upgrading from Stretch to Buster

Currently the app relies on a custom install of PHP 7.1 / 7.2 / 7.3 which should be adapted. Imho (to be discussed and see with other app packagers after we gain more XP with this) : 
- the scripts should be adapted to use PHP 7.3 instead of 7.2
- the upgrade script should take care of the move from 7.2 to 7.3 (which for users could and should therefore happen prior to the upgrade to buster)
- we should have an if/else to install PHP 7.3 only if we're on stretch (and not on buster) [similar to what's done for pixelfed for example](https://github.com/YunoHost-Apps/pixelfed_ynh/pull/102)

### PR DESCRIPTION
**Pull request by SimonMellerin**
Fix #26 

[ Build Status&build=2)
](https://ci-apps-dev.yunohost.org/jenkins/job/pilea_ynh%20(SimonMellerin)/2/)

@yalh76 CI Results for Revision: cbac92245ab051676127d0839ded72a16959a8b1 :
### Stretch;
image

### Buster:
image

### CODE REVIEW

**Comment by SimonMellerin** [comment]
[Build Status&build=2)](https://ci-apps-dev.yunohost.org/jenkins/job/pilea_ynh%20(SimonMellerin)/2/)

**Review by yalh76 — APPROVED** [review]

File: scripts/install [hunk 1]
```diff
@@ -1,117 +1,94 @@
...
 #=================================================
 # NGINX CONFIGURATION
 #=================================================
-ynh_script_progression --message="Configuring nginx web server..." --time --weight=1
+ynh_script_progression --message="Configuring nginx web server..." --weight=1
 
+# Create a dedicated nginx config
 ynh_add_nginx_config
+
 if [ "$path_url" = "/" ]
```
Comment (SimonMellerin) [inline scripts/install:92]: This part is not in the example package, is it still necessary ? What do you think @yalh76 ?

File: scripts/backup [hunk 2]
```diff
@@ -18,45 +20,37 @@ ynh_abort_if_errors
 #=================================================
 # LOAD SETTINGS
 #=================================================
-ynh_script_progression --message="Loading installation settings..." --time --weight=1
+ynh_script_progression --message="Loading installation settings..." --weight=1
```
Comment (SimonMellerin) [inline scripts/backup:23]: I've just launched package_linter, why did you delete all these '--time' option ? 
The package_linter seems not happy without them :(

File: scripts/install (hunk 1 above)
Comment (yalh76) [reply scripts/install:92]: I'm not sure it's necessary but haven't the time to test without, so I keep it

File: scripts/backup (hunk 2 above)
Comment (yalh76) [reply scripts/backup:23]: As indicate in example_ynh:

https://github.com/YunoHost/example_ynh/blob/429b410dc591626b99b7972796f2dbfd1be0b040/scripts/install#L49-L56

--time is a packager option, This option should be removed before releasing your app.

You only keep it during one package_check to be able to provide a `weigth` value for each ynh_script_progression

So if you want to adjust the weight of each `ynh_script_progression`, you can add them back and also `--last` for the last  `ynh_script_progression`, launch a package check, retrieve the value of each step, adjust the `weight` for each  `ynh_script_progression` and remove `--time` and `--last`

Comment (SimonMellerin) [reply scripts/backup:23]: Ok thanks !
